From ad6b301c0c1e5852f0d1efa6402cc9b8a2b52ff3 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 18 Jul 2019 23:23:53 +0300
Subject: [PATCH 1/1] Avoid using built-in cursors.

This changes our cursor setting for LEFT_PTR to use from_pixbuf, using a pixbuf of the "default" cursor icon from breeze-cursor icons.
---
 src/slick-greeter.vala | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/slick-greeter.vala b/src/slick-greeter.vala
index 8d4ef8d..ffeca43 100644
--- a/src/slick-greeter.vala
+++ b/src/slick-greeter.vala
@@ -637,9 +637,12 @@ public class SlickGreeter
 
         debug ("Starting slick-greeter %s UID=%d LANG=%s", Config.VERSION, (int) Posix.getuid (), Environment.get_variable ("LANG"));
 
-        /* Set the cursor to not be the crap default */
-        debug ("Setting cursor");
-        Gdk.get_default_root_window ().set_cursor (new Gdk.Cursor.for_display (Gdk.Display.get_default (), Gdk.CursorType.LEFT_PTR));
+        try {
+            var cursor = new Gdk.Pixbuf.from_file("/usr/share/pixmaps/slick-greeter-pointer.png");
+            Gdk.get_default_root_window().set_cursor(new Gdk.Cursor.from_pixbuf(Gdk.Display.get_default(),cursor, 0, 0));
+        } catch (Error e) {
+            Gdk.get_default_root_window ().set_cursor (new Gdk.Cursor.for_display (Gdk.Display.get_default (), Gdk.CursorType.LEFT_PTR));
+        }
 
         bool do_show_version = false;
         bool do_test_mode = false;
-- 
2.22.0

